mono/metadata/threads.c

```diff
@@ -5881,4 +5882,131 @@ mono_thread_internal_is_current (MonoInternalThread *internal)
 void
 mono_set_thread_dump_dir (gchar* dir) {
 	thread_dump_dir = dir;
-}
\ No newline at end of file
+}
+
+#ifndef HOST_WIN32
+
+static size_t num_threads_summarized = 0;
+
+// mono_thread_internal_current doesn't always work in signal
+// handler contexts. This does.
+static MonoInternalThread *
+find_missing_thread (MonoNativeThreadId id)
+{
+	MonoInternalThread *thread_array [128];
+	int nthreads = collect_threads (thread_array, 128);
+
+	for (int i=0; i < nthreads; i++) {
+		MonoNativeThreadId tid = thread_get_tid (thread_array [i]);
+		if (tid == id)
+			return thread_array [i];
+	}
+	return NULL;
+}
+
+static gboolean
+mono_threads_summarize_one (MonoThreadSummary *out, MonoContext *ctx)
+{
+	MonoDomain *domain;
+
+	MonoNativeThreadId current = mono_native_thread_id_get();
+	MonoInternalThread *thread = find_missing_thread (current);
+
+	// Not one of ours
+	if (!thread)
+		return FALSE;
+
+	memset (out, 0, sizeof (MonoThreadSummary));
```

Comment:
sizeof (*out)?

Reply:
*I* think that looks really strange. But I'll do it. It should be functionally identical here.